WFPREV-222

yzlucas opened the pull request:
Add all activity polygon to the fiscal map
Attach fiscal year to the polygons
Display the project coordinate pin (if there is any) when no polygons are available.
Adding legend to the map
Add unit tests 
![image](https://github.com/user-attachments/assets/16de790a-678b-4612-855c-a48a02f97ae7)
![image](https://github.com/user-attachments/assets/876f9ea7-64ae-48ef-8d7b-465090d0cd72)

ssylver93 reviewed (changes requested):
Code looks good, one change I would suggest however: the legend in your screenshots look like there is no padding on the left side of the title and icons within the legend div, can you add some padding there to match figma?
![image](https://github.com/user-attachments/assets/aa8ecf3f-c306-4f7e-98be-c099fb26f72f)

yzlucas commented:
> Code looks good, one change I would suggest however: the legend in your screenshots look like there is no padding on the left side of the title and icons within the legend div, can you add some padding there to match figma? ![image](https://private-user-images.githubusercontent.com/107515688/431534858-aa8ecf3f-c306-4f7e-98be-c099fb26f72f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2MDg5OTQsIm5iZiI6MTc0NDYwODY5NCwicGF0aCI6Ii8xMDc1MTU2ODgvNDMxNTM0ODU4LWFhOGVjZjNmLWMzMDYtNGY3ZS05OGJlLWMwOTlmYjI2ZjcyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxNFQwNTMxMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNWMyODIwY2JiMjlkNWJmNjNlNjVmNWRjNjRkM2JlMjVjOTIzODhiNGM2MGI0ZGQ5NDkxZTQwNzM0ZmQ1NTUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CLU4eybpZmw-61dzdYd_RJDm0zU6PAoL7xBfsBnt_E8)

Resolved, thanks!
![image](https://github.com/user-attachments/assets/0c83cbec-5a37-43aa-aca7-65815eeaf4ee)

ssylver93 commented on client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts:
```diff
@@ -0,0 +1,231 @@
+    this.addLegend();
+  }
+
+  addLegend(): void {
```
Can this function be put in a common file or service so it can be used on the map and project and details pages?

yzlucas replied:
added to tools.ts

ssylver93 commented on server/wfprev-api/pom.xml:
```diff
@@ -13,6 +13,15 @@
     <artifactId>wfprev-api</artifactId>
     <version>1.0.0-SNAPSHOT</version>
+    <repositories>
+        <repository>
```
These changes aren't needed, they can be removed

yzlucas replied:
removed

ssylver93 commented on server/wfprev-api/pom.xml:
```diff
@@ -222,6 +231,23 @@
             <version>2.0.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.geotools</groupId>
```
Please remove

yzlucas replied:
removed

ssylver93 commented on server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/services/ProjectBoundaryService.java:
```diff
@@ -23,6 +23,13 @@
 import java.util.UUID;
 
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Geometry;
```
Please remove, we don't need these dependencies at this point

yzlucas replied:
removed

ssylver93 reviewed (changes requested):
1 comment about reusing the map legend and some requests to remove dependencies in API

yzlucas commented:
> 1 comment about reusing the map legend and some requests to remove dependencies in API

resolved, 
also fixed failing unit tests after mege, added full map navigation 

![image](https://github.com/user-attachments/assets/357d3f95-9612-4062-bed0-d83ca9b46ba8)

dhlevi commented on client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts:
```diff
@@ -0,0 +1,318 @@
+        color = this.fiscalColorMap.future;
+      }
+  
+      boundaryEntry.boundary.forEach((item: any) => {
```
It is almost always better to use `for(let thing of things)` vs an `array.forEach`. It's generally clearer to read and debug, and also performs better

yzlucas replied:
Sure, changed to use for loop

dhlevi commented on client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts:
```diff
@@ -0,0 +1,318 @@
+      const delta = 0.01;
+  
+      const bbox = [
+        (lng - delta).toFixed(6),
```
Any specific reason for fixing to 6?

yzlucas replied:
Just concern about the coordinate has extremely long decimals.  But just confirmed when user add coordinate via Create Project, it is already being handled. So I will remove this.

dhlevi commented on client/wfprev-war/src/main/angular/src/app/components/map/map.component.ts:
```diff
@@ -32,20 +35,46 @@ export class MapComponent implements AfterViewInit {
+          
+                const smk = this.mapService.getSMKInstance();
+                debugger
+                if (smk?.$viewer?.map?.fitBounds) {
```
While the `?` operator works well, be careful of overuse as it's simple, remember that you need to handle situations where that viewer or map object may be null/undefined, and how you want to handle that vs. just puking an error

yzlucas replied:
Wrapped this in try catch block, and added fallback return checks for bboxParam and bboxParts

dhlevi reviewed (dismissed):
Overall looks OK, some comments that would be nice to address or learn from for the future.
